samples/go/photo-dedup/model/photo.go

```diff
@@ -0,0 +1,73 @@
+// Copyright 2016 Attic Labs, Inc. All rights reserved.
+// Licensed under the Apache License, version 2.0:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package model
+
+import (
+	"github.com/attic-labs/noms/go/types"
+	"github.com/attic-labs/noms/samples/go/photo-dedup/dhash"
+)
+
+type Photo interface {
+	ID() ID
+	Dhash() dhash.Hash
+	SetDhash(dhash.Hash)
+	IterSizes(cb func(width int, height int, url string))
+	Marshal() types.Struct
+}
+
+type lazyPhoto struct {
```

Comment:
Here's the problem: 

If I unmarshal into a go Photo struct that doesn't define all the fields in the noms struct, those fields will be lost when marshalling back into a noms struct.

Any job that is adding a field to a struct will have this problem. Not sure what the best solution to this problem is yet. One would be to have an option that forces the go struct to define all the fields in the noms struct. This would fix the problem, but it means that job code has to change whenever somethings added to the schema. Another solution would be to provide a way to merge go struct into a noms struct on unmarshal.